### ISSUE TITLE
[REQUEST] Scrape for Cloud inspection files first

### ISSUE THREAD

**Issue by LoganWolverineX** [nixgates/plugin.video.seren#284]
**Is your feature request related to a problem? Please describe.**
I know this feature request is not related to a problem.  It was something I noticed pre-2.0 and wait until 2.0 was out to see if it was added/changed. 

**Describe the solution you'd like**
When “inspect cloud files on scrape” is enabled for any cloud services, it should be the first thing it should try to scan. Ive noticed it’s near the end of the scrapping process. 

**Describe alternatives you've considered**
Alternative is to just wait for it to come to the cloud inspection part of scrapping. The problem then comes to the scraping time out rate. Having this in the latter part of scrapping may affect scrapping time which comes back empty. 

Another route is maybe count cloud inspection separately and have its own timeout rate instead of as part of the timeout rate from scrapping links for other sites.

**Comment by Desperados85** [comment]
+1 for this! 
In the original Seren I had a problem playing my RD cloud files. When I scraped a movie, my cloud file would appear on top but everytime I tried to play it, it said no playable sources found. 
So when Seren 2.0 came out that's the first thing I wanted to test. But now it doesn't even scrape my cloud it seems. I've tried several movies that I know are in my cloud but it never shows my cloud files.. Didn't think about checking the time out rate since it worked before the update to 2.0. 
But it would be great if cloud sources could be scraped first, Fen does this so I know it's possible :)

**Comment by 5ilv3r5urf3r** [comment]
Another vote for this.

**Comment by nixgates** [comment]
@Desperados85 @LoganWolverineX @5ilv3r5urf3r 

could I ask for one of you guys to test this in 2.0.8

I know I fixed some issues with episode identification which may have played a part in this problem

**Comment by LoganWolverineX** [comment]
> @Desperados85 @LoganWolverineX @5ilv3r5urf3r
> 
> could I ask for one of you guys to test this in 2.0.8
> 
> I know I fixed some issues with episode identification which may have played a part in this problem

Testing now.  I don't need to grab from github master do i?

**Comment by LoganWolverineX** [comment]
@nixgates , i just tested it.  It now is able to script for PM cloud, but not for RD cloud.

**Comment by LoganWolverineX** [comment]
@nixgates : so to summarize my test.

- PM Cloud Inspection works now
- RD cloud inspection still does not work and is not scraped in the scraping results.
- cloud inspection still isn't the first category to scrape.  it's the last category to scrape.

**Comment by nixgates** [comment]
@LoganWolverineX not sure what you mean first category to scrape? All providers including cloud occur at the same time.

**Comment by LoganWolverineX** [comment]
@nixgates for this enhancement request, i was asking if you could make the cloud inspection come first before searching for the other providers.  I noticed that pre 2.0 version, enabling preemptive termination sometimes causes it to end scraping too soon which misses the cloud inspection part of the searching.

The bug where PM and RD cloud inspection not working is for issue #278 which i didn't want to you get confused between the two items.

**Comment by nixgates** [comment]
@LoganWolverineX Ah ok, that's a possible feature. But, at the moment I'd just like to focus on core issues so I'll just stick to the RD Cloud inspection fix and we can look at adding that optional feature after things cool down again :)

**Comment by LoganWolverineX** [comment]
@nixgates yep, i completely understand it.  i just didn't want you to get confused between my two entries i posted :).  I just wanted to add it in so i don't forget to put it in later.  yea, feature in this request (#284) is definitely low priority while issue #278 is a core issue.

**Comment by nixgates** [comment]
Ah apologies! You did everything possible to make sure I didn't get confused and yet I did! lol

Sorry about that 😞

**Comment by LoganWolverineX** [comment]
Haha! It’s all good. I should have just waited to just post the feature request until later to avoid confusion. So partially my fault 🤣. Thanks for all that you do for the addon!

**Comment by LoganWolverineX** [comment]
@nixgates just wanted to make sure "awaiting update" should not be tagged to this enhancement request, right? it will be looked at a later date.

**Comment by LoganWolverineX** [comment]
I don’t want this ticket closed until the new update is out so I’m just keeping it active for now.

**Comment by LoganWolverineX** [comment]
Pinged again to remove stale status

**Comment by drinfernoo** [comment]
Scraping basically occurs for all providers in parallel.

If some providers finish first, then that means they're just finishing faster... not that they started first.

**Comment by LoganWolverineX** [comment]
i understand all providers scrape in parallel, but yes, i was hoping to request that somehow it can be implemented that cloud files for debrid services are scraped first before other providers.  that would be nice since most of the time cloud files are the ones that we want to stream.

**Comment by LoganWolverineX** [comment]
ping so it doesn't close
